source/solvers/navier_stokes_base.cc

```diff
@@ -117,6 +119,18 @@ NavierStokesBase<dim, VectorType, DofsType>::NavierStokesBase(
   this->pcout.set_condition(
     Utilities::MPI::this_mpi_process(this->mpi_communicator) == 0);
 
+  // Check if the output directory exists
+  std::string output_dir_name =
+    simulation_parameters.simulation_control.output_folder;
+  struct stat buffer;
+
+  // If output directory does not exist, create it
+  if (stat(output_dir_name.c_str(), &buffer) != 0 &&
+      Utilities::MPI::this_mpi_process(this->mpi_communicator) == 0)
+    {
+      create_output_folder(output_dir_name);
+    }
```

Comment:
I would put the this->mpi_process ==0 in a first if so that only proc 0 checks if the folder exists. at 1000 cores otherwise you will have 1000 processes checking the same folder.

Reply:
done.